File: backend/app/models/project.py
"""Project model."""
import uuid
from datetime import datetime
from sqlalchemy import String, Integer, Text, DateTime, func
from sqlalchemy.orm import Mapped, mapped_column
from app.database import Base


class Project(Base):
    __tablename__ = "projects"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    title: Mapped[str] = mapped_column(String(500), default="未命名项目")
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    genre: Mapped[str] = mapped_column(String(100), default="玄幻")
    target_words: Mapped[int] = mapped_column(Integer, default=100000)
    narrative_perspective: Mapped[str] = mapped_column(String(50), default="第三人称")
    status: Mapped[str] = mapped_column(String(50), default="planning")  # planning/writing/revising/completed
    total_word_count: Mapped[int] = mapped_column(Integer, default=0)
    outline_mode: Mapped[str] = mapped_column(String(50), default="one-to-one")  # one-to-one / one-to-many
    world_setting: Mapped[str | None] = mapped_column(Text, nullable=True)  # JSON string
    cover_prompt: Mapped[str | None] = mapped_column(Text, nullable=True)
    cover_url: Mapped[str | None] = mapped_column(String(2000), nullable=True)
    writing_style_id: Mapped[str | None] = mapped_column(String(36), nullable=True)
    prompt_template_id: Mapped[str | None] = mapped_column(String(36), nullable=True)
    active_skill: Mapped[str | None] = mapped_column(String(100), nullable=True)
    generation_config: Mapped[str | None] = mapped_column(Text, nullable=True)  # JSON: {provider, model, api_key, ...}
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now(), onupdate=func.now())

File: frontend/src/types/project.ts
export interface ProjectSummary {
  id: string; title: string; description: string | null; genre: string;
  status: string; total_word_count: number; narrative_perspective: string;
  outline_mode: string; cover_url: string | null; created_at: string; updated_at: string;
}
export interface Project {
  id: string; title: string; description: string | null; genre: string;
  target_words: number; narrative_perspective: string; status: string;
  total_word_count: number; outline_mode: string;
  world_setting: Record<string, unknown> | null; cover_prompt: string | null;
  cover_url: string | null; writing_style_id: string | null;
  active_skill: string | null; generation_config: Record<string, unknown> | null;
  created_at: string; updated_at: string;
}
export interface CreateProjectData {
  title: string; description?: string; genre?: string; target_words?: number;
  narrative_perspective?: string; outline_mode?: string;
  world_setting?: Record<string, unknown>; writing_style_id?: string;
  generation_config?: Record<string, unknown>;
}
